### PR TITLE
ui: add market maker stats to markets view

### PR DESCRIPTION
**Pull request by buck54321**
When market maker is running and we are on a market fora configured bot in the markets view, show market maker stats.
image

### CODE REVIEW

File: client/webserver/site/src/html/markets.tmpl [hunk 1]
```diff
@@ -263,13 +263,139 @@
...
+
+                    <div data-tmpl="pendingDepositBox" class="d-flex justify-content-between align-items-center">
+                      <span>[[[Pending deposits]]]</span>
+                      <span data-tmpl="pendingDeposits"></span>
+                    </div>
+
+                    <div data-tmpl="pendingWithdrawalBox" class="d-flex justify-content-between align-items-center">
+                      <span>[[[Pending withdrawals]]]</span>
+                      <span data-tmpl="pendingWithdrawals"></span>
+                    </div>
```
Comment (martonp) [inline client/webserver/site/src/html/markets.tmpl:354]: Pending deposits/withdrawals should be hidden when basic bot is running.

File: client/webserver/site/src/html/markets.tmpl [hunk 2]
```diff
@@ -263,13 +263,139 @@
...
+                      </span>
+                    </div>
+
+                    <div data-tmpl="roundTripFeesBox" class="d-flex justify-content-between align-items-center">
+                      <span>[[[Round_trip fees]]]</span>
+                      <span>
+                        <span data-tmpl="roundTripFeesUSD"></span>
+                        <span class="fs14 grey">USD</span>
+                      </span>
+                    </div>
```
Comment (martonp) [inline client/webserver/site/src/html/markets.tmpl:394]: I think a link to the `/mmlogs` page for this run would be good. These stats could be over there as well. I think the info about the market i.e. basis price, fee gap etc. would be more useful here, and then data about the run could be over there.

File: client/webserver/site/src/html/markets.tmpl [hunk 3]
```diff
@@ -263,13 +263,139 @@
...
+                        <span data-tmpl="feeGap"></span> (<span data-tmpl="feeGapPct"></span>%)
+                      </span>
+                    </div>
+
+                    <div data-tmpl="remoteGapBox" class="d-flex justify-content-between align-items-center">
+                      <span>[[[Remote gap]]]</span>
+                      <span>
+                        <span data-tmpl="remoteGap"></span> (<span data-tmpl="remoteGapPct"></span>%)
+                      </span>
+                    </div>
```
Comment (martonp) [inline client/webserver/site/src/html/markets.tmpl:386]: I think fee gap and remote gap could use a tooltip.

**Review by martonp — COMMENTED** [review]
It looks cool. It keeps flashing though.. there is a big flash when the first order gets placed, and then there are constantly flashes like in the video. It happens at 0:13.

https://github.com/decred/dcrdex/assets/6186350/08bc4ec8-58a4-43fe-96a5-22d8127124b6

**Review by JoeGruffins — COMMENTED** [review]
Trading on simnet with bots and seeing some strange numbers. Also seeing the flashing like @martonp 

image

image

File: client/webserver/site/src/html/markets.tmpl [hunk 4]
```diff
@@ -263,13 +263,139 @@
...
+                      <span>[[[Pending withdrawals]]]</span>
+                      <span data-tmpl="pendingWithdrawals"></span>
+                    </div>
+
+                    <div class="d-flex justify-content-between align-items-center">
+                      <span>[[[Settled matches]]]</span>
+                      <span data-tmpl="completedMatches"></span>
+                    </div>
+
+                    <div data-tmpl="tradedUSDBox" class="d-flex justify-content-between align-items-center">
```
[the remark this answers is not in the document]
Comment (buck54321) [reply client/webserver/site/src/html/markets.tmpl:361]: It's a measure of how much action the bot has seen.

**Review by martonp — APPROVED** [review]

**Comment by martonp** [comment]
@buck54321 I still see the Pending Deposit and Pending Withdrawals when the basic mm bot is running. Otherwise, it looks great.

**Review by JoeGruffins — APPROVED** [review]

**Comment by JoeGruffins** [comment]
nit: Percentages could be rounded here:
image

**Comment by JoeGruffins** [comment]
Unrelated to these changes, but it looks like it is possible to miss a cancel if you are changing bot settings and one of the orders in in epoch, then the bot will let the order sit forever.

**Pull request merged**